File: leetcode/p204.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 3:
            return 0
        primes = [True] * n
        primes[0] = False
        primes[1] = False
        for i in range(2, int(n**0.5+1)):
            if not primes[i]:
                continue
            t = i+i
            while t < n:
                primes[t] = False
                t += i
        return sum(primes)

File: leetcode/p204.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n<3) return 0;
        vector<bool> table(n, true);
        table[0] = false;
        table[1] = false;
        int total = n-2;
        for (int i=2;i<sqrt(n)+1;i++) {
            if (!table[i]) continue;
            for (int j=i+i;j<n;j+=i) {
                if (table[j]) {
                    table[j] = false;
                    total--;
                }
            }
        }
        return total;
    }
};
